dtsync/publisher.go

```diff
@@ -10,8 +10,8 @@ import (
 	"time"
 
 	dt "github.com/filecoin-project/go-data-transfer"
-	"github.com/filecoin-project/go-legs/gpubsub"
 	"github.com/filecoin-project/go-legs/p2p/protocol/head"
+	"github.com/filecoin-project/storetheindex/announce/gossiptopic"
```

Comment:
This PR fixes some defects as well as better organized the code, so I want go-legs to use the new announce handling code. There are a few options:
- Copy the `announce` package code into go-legs, so that it is present in both storetheindex and go-legs. Then port the other parts of go-legs into storetheindex and sever any dependency between the two.
- Put the `announce` package into some other repo that both go-legs and storetheindex can depend on.
- Move the `announce` package into go-legs and have storetheindex (coordination service) import the `announce` package from go-legs until.  ...and sometime in the future port go-legs into storetheindex.